tietokanta/src/main/resources/db/migration/V1_483__.sql

```diff
@@ -0,0 +1,91 @@
+-- Irrota päällystyslomakkeen määrämuutokset omaan tauluun
+
+CREATE TYPE maaramuutos_tyon_tyyppi AS ENUM ('ajoradan_paallyste', 'pienaluetyot', 'tasaukset', 'jyrsinnat',
+                                              'muut');
+
+CREATE TABLE yllapitokohteen_maaramuutos (
+  id serial PRIMARY KEY,
+  yllapitokohde INTEGER REFERENCES yllapitokohde (id) NOT NULL,
+  tyon_tyyppi maaramuutos_tyon_tyyppi NOT NULL,
+  tyo VARCHAR(256) NOT NULL,
+  yksikko VARCHAR(32) NOT NULL,
+  tilattu_maara NUMERIC NOT NULL,
+  toteutunut_maara NUMERIC NOT NULL,
+  yksikkohinta NUMERIC NOT NULL,
+  poistettu boolean DEFAULT FALSE NOT NULL,
+  luoja INTEGER REFERENCES kayttaja (id) NOT NULL,
+  luotu TIMESTAMP DEFAULT NOW()  NOT NULL,
+  muokkaaja INTEGER REFERENCES kayttaja (id),
+  muokattu TIMESTAMP
+);
+
+ALTER TABLE paallystysilmoitus DROP COLUMN muutoshinta; -- Lasketaan jatkossa yllä olevasta taulusta
+ALTER TABLE paallystysilmoitus DROP COLUMN paatos_taloudellinen_osa; -- Hinnanmuutosten hyväksyminen jää pois (HAR-4090)
+ALTER TABLE paallystysilmoitus DROP COLUMN perustelu_taloudellinen_osa;
+ALTER TABLE paallystysilmoitus DROP COLUMN kasittelyaika_taloudellinen_osa;
+ALTER TABLE paallystysilmoitus DROP COLUMN asiatarkastus_taloudellinen_osa;
+
+-- Migratoi olemassa olevien päällystysilmoitusten ilmoitustiedot-JSONista taloudellisen osan
+-- tiedot uuteen tauluun
+
+CREATE OR REPLACE FUNCTION muunna_paallystysilmoitusten_maaramuutokset() RETURNS VOID AS
+$BODY$
+DECLARE
+  rivi RECORD;
+BEGIN
+  FOR rivi IN SELECT
+                paallystyskohde,
+                luoja,
+                json_array_elements((ilmoitustiedot->'tyot')::JSON) AS tyo
+              FROM paallystysilmoitus
+  LOOP
+    INSERT INTO yllapitokohteen_maaramuutos (yllapitokohde,
+                                             tyon_tyyppi,
+                                             tyo,
+                                             yksikko,
+                                             tilattu_maara,
+                                             toteutunut_maara,
+                                             yksikkohinta,
+                                             luoja)
+    VALUES (rivi.paallystyskohde,
+            CASE
+              WHEN (rivi.tyo->>'tyyppi')='ajoradan-paallyste' THEN 'ajoradan_paallyste'::maaramuutos_tyon_tyyppi
+              ELSE (rivi.tyo->>'tyyppi')::maaramuutos_tyon_tyyppi -- Kaikki muut voi käsitellä sellaisenaan
+            END,
+            rivi.tyo->>'tyo',
+            rivi.tyo->>'yksikko',
+            (rivi.tyo->>'tilattu-maara')::INTEGER,
+            (rivi.tyo->>'toteutunut-maara')::INTEGER,
+            (rivi.tyo->>'yksikkohinta')::INTEGER,
+            rivi.luoja);
+  END LOOP;
+  RETURN;
+END
+$BODY$
+LANGUAGE 'plpgsql';
+
+SELECT * FROM muunna_paallystysilmoitusten_maaramuutokset();
```

Comment:
Note to self: Nämä pystyy muistaakseni ajamaan myös PERFORM, jos haluaa ilman turhaa selectiä..